File: assignments/ruby/pig-latin/pig-latin_test.rb
require 'minitest/autorun'
require 'minitest/pride'
require_relative 'pig_latin'

class PigLatinTest < MiniTest::Unit::TestCase

  def test_word_beginning_with_a
    assert_equal "appleay", PigLatin.translate("apple")
  end

  def test_other_word_beginning_e
    skip
    assert_equal "earay", PigLatin.translate("ear")
  end

  def test_word_beginning_with_p
    skip
    assert_equal "igpay", PigLatin.translate("pig")
  end

  def test_word_beginning_with_k
    skip
    assert_equal "oalakay", PigLatin.translate("koala")
  end

  def test_word_beginning_with_ch
    skip
    assert_equal "airchay", PigLatin.translate("chair")
  end

  def test_word_beginning_with_qu
    skip
    assert_equal "eenquay", PigLatin.translate("queen")
  end

  def test_word_with_consonant_preceding_qu
    skip
    assert_equal "aresquay", PigLatin.translate("square")
  end

  def test_word_beginning_with_th
    skip
    assert_equal "erapythay", PigLatin.translate("therapy")
  end

  def test_word_beginning_with_thr
    skip
    assert_equal "ushthray", PigLatin.translate("thrush")
  end

  def test_word_beginning_with_sch
    skip
    assert_equal "oolschay", PigLatin.translate("school")
  end

  def test_translates_phrase
    skip
    assert_equal "ickquay astfay unray", PigLatin.translate("quick fast run")
  end

end

File: assignments/haskell/pig-latin/pig-latin_test.hs
import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
import System.Exit (ExitCode(..), exitWith)
import PigLatin (translate)

exitProperly :: IO Counts -> IO ()
exitProperly m = do
  counts <- m
  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess

testCase :: String -> Assertion -> Test
testCase label assertion = TestLabel label (TestCase assertion)

main :: IO ()
main = exitProperly $ runTestTT $ TestList
       [ TestList pigLatinTests ]

pigLatinTests :: [Test]
pigLatinTests =
  [ testCase "beginning with vowels" $ do
    "appleay" @=? translate "apple"
    "earay" @=? translate "ear"
  , testCase "beginning with single letter consonant clusters" $ do
    "igpay" @=? translate "pig"
    "oalakay" @=? translate "koala"
    "atqay"   @=? translate "qat"
  , testCase "beginning with multiple letter consonant clusters" $ do
    "airchay" @=? translate "chair"
    "erapythay" @=? translate "therapy"
    "ushthray" @=? translate "thrush"
    "oolschay" @=? translate "school"
  , testCase "consonant clusters with qu" $ do
    "eenquay" @=? translate "queen"
    "aresquay" @=? translate "square"
  , testCase "phrase" $
    "ickquay astfay unray" @=? translate "quick fast run"
  ]
